### PR TITLE
Add support for new async validation endpoints

### PR DESCRIPTION
**Pull request by huluvu21**
### Motivation

We want to cover the new async project validation endpoints in their raw form. The PR can be merger after the feature is released.

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

It would be great to test those endpoints after the release to production to make sure.

### CODE REVIEW

File: Kontent.Ai.Management/IManagementClient.cs [hunk 1]
```diff
@@ -568,6 +570,24 @@ public interface IManagementClient
     /// <returns><see cref="ProjectReportModel"/></returns>
     Task<ProjectReportModel> ValidateProjectAsync();
 
+    /// <summary>
+    /// Initiates asynchronous project validation.
+    /// </summary>
+    /// <returns><see cref="AsyncValidationTask"/></returns>
+    Task<AsyncValidationTask> InitiateProjectAsyncValidationTaskAsync();
```
Comment (gormal) [inline Kontent.Ai.Management/IManagementClient.cs:577]: ManagementClient should return *Model

File: Kontent.Ai.Management/IManagementClient.cs [hunk 2]
```diff
@@ -568,6 +570,24 @@ public interface IManagementClient
...
+
+    /// <summary>
+    /// Gets async validation task.
+    /// </summary>
+    /// <returns><see cref="AsyncValidationTask"/></returns>
+    Task<AsyncValidationTask> GetAsyncValidationTaskAsync(Guid taskId);
```
Comment (gormal) [inline Kontent.Ai.Management/IManagementClient.cs:583]: ManagementClient should return *Model

File: Kontent.Ai.Management/IManagementClient.cs [hunk 3]
```diff
@@ -568,6 +570,24 @@ public interface IManagementClient
...
+
+    /// <summary>
+    /// Lists async validation task issues.
+    /// </summary>
+    /// <returns><see cref="IListingResponseModel{AsyncValidationTaskIssue}"/></returns>
+    Task<IListingResponseModel<AsyncValidationTaskIssue>> ListAsyncValidationTaskIssuesAsync(Guid taskId);
```
Comment (gormal) [inline Kontent.Ai.Management/IManagementClient.cs:589]: ManagementClient should return *Model

**Review by gormal — CHANGES_REQUESTED** [review]
Changes look great! There are missing tests for endpointUrlBuilder and a few nitpicks.

**Comment by huluvu21** [comment]
> Changes look great! There are missing tests for endpointUrlBuilder and a few nitpicks.

All right. Done (fingers crossed).

**Review by gormal — APPROVED** [review]

**Pull request merged**